src/views/forms/TriggersForm/LimitPriceInputs.tsx

```diff
@@ -15,26 +21,80 @@ import { FormInput } from '@/components/FormInput';
 import { Tag } from '@/components/Tag';
 import { WithTooltip } from '@/components/WithTooltip';
 
+import { getTriggerOrdersInputs } from '@/state/inputsSelectors';
+
+import abacusStateManager from '@/lib/abacus';
+import { MustBigNumber } from '@/lib/numbers';
+
 type ElementProps = {
+  existsLimitOrder: boolean;
+  multipleTakeProfitOrders: boolean; 
+  multipleStopLossOrders: boolean;
   tickSizeDecimals?: number;
 };
 
 type StyleProps = {
   className?: string;
 };
 
-export const LimitPriceInputs = ({ tickSizeDecimals, className }: ElementProps & StyleProps) => {
+export const LimitPriceInputs = ({
+  existsLimitOrder,
+  multipleTakeProfitOrders,
+  multipleStopLossOrders,
+  tickSizeDecimals,
+  className,
+}: ElementProps & StyleProps) => {
   const stringGetter = useStringGetter();
 
+  const { stopLossOrder, takeProfitOrder } =
+    useSelector(getTriggerOrdersInputs, shallowEqual) || {};
+
   const [shouldShowLimitPrice, setShouldShowLimitPrice] = useState(false);
 
+  const decimals = tickSizeDecimals ?? USD_DECIMALS;
+
+  useEffect(() => {
+    setShouldShowLimitPrice(existsLimitOrder);
+  }, [existsLimitOrder]);
+
+  const onToggleLimit = (limitToggled: boolean) => {
```

Comment:
also above question re resetting limit price and price disappearing

Reply:
> also right now when i toggle limit price, it will disappear if i uncheck and check it, is that expected?

yup! So unchecking it is implicitly saying that you want to revert to setting the order to be a market order - if you re-check it, you have to re- add the values.